Don't always use VK_KHR_surface_capabilities2 when trimming

mikes-lunarg opened the pull request:
Capture was automatically upgrading trimmed vkGetPhysicalDeviceSurfaceCapabilitiesKHR calls into vkGetPhysicalDeviceSurfaceCapabilities2KHR calls, whether or not the VK_KHR_surface_capabilities2 extension was actually enabled. This causes a replay validation error (and a crash when replayed through RenderDoc) if the original capture did not use VK_KHR_surface_capabilities2. The fix is to track which version of the capability and format queries was captured and then write the same version to the trimmed capture file.

Fixes #2247

ci-tester-lunarg commented:
CI gfxreconstruct build queued with queue ID 472836.

ci-tester-lunarg commented:
CI gfxreconstruct build # 7105 running.

ci-tester-lunarg commented:
CI gfxreconstruct build # 7105 passed.

panos-lunarg commented on framework/encode/vulkan_handle_wrappers.h:
```diff
@@ -517,25 +517,37 @@ struct CommandPoolWrapper : public HandleWrapper<VkCommandPool>
-    VkSurfaceCapabilities2KHR  surface_capabilities;
-    std::unique_ptr<uint8_t[]> surface_capabilities_pnext_memory;
+    // For vkGetPhysicalDeviceSurfaceCapabilities2KHR
+    bool use_surface_capabilities2{ false };
```
Is it possible that we need to differentiate even between `2` and `2KHR`?

I was doing a similar change for `vkCreateRenderPass` and it made a difference whether the application was calling `2` and I latter called `2KHR` with getting a warning `Unsupported function vkCreateRenderPass2KHR was called, resulting in no-op behavior`.

panos-lunarg replied:
doh, you're right

ci-tester-lunarg commented:
CI gfxreconstruct build queued with queue ID 477385.

ci-tester-lunarg commented:
CI gfxreconstruct build # 7163 running.

mikes-lunarg commented:
Rebased and implemented @bradgrantham-lunarg's `std::variant` suggestion from https://github.com/LunarG/gfxreconstruct/pull/2250#discussion_r2169334129

ci-tester-lunarg commented:
CI gfxreconstruct build # 7164 running.
